Use function pointers to call functions on all platforms

DavidCurtiss opened the pull request:
### What does this Pull Request accomplish?

Uses loaded function pointers on Windows, too, where we had been using delayload'ed symbols.

### Why should this Pull Request be merged?

Consistency, simplicity, and [AB#2209410](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2209410)

### What testing has been done?

Using an equivalent code change, tested the [AB#2209410](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2209410) use case.

bkeryan commented:
Is this a temporary workaround or a permanent change?

If it's a temporary workaround, you could do something like `#if defined(_MSC_VER) && USE_DELAY_LOAD`.

If it's a permanent change, you may be able to delete the import libraries from disk and remove `CMakeLists.txt`'s references to `Delayimp`, the import libraries, and `/DELAYLOAD`.

DavidCurtiss commented:
> Is this a temporary workaround or a permanent change?

It's permanent, so I've now deleted the delayload stuff and the import libraries.

I also added some compilation tests, because, as @astarche previously pointed out to me over Teams, the former implementation was providing "compile time validation that we got the [function parameter] types ... close enough to compile against the header."

astarche reviewed (approved):
Looks great! Multiple nice improvements. Thanks for sorting out the details.

reckenro commented on source/codegen/templates/compilation_test.cpp.mako:
```diff
@@ -0,0 +1,36 @@
+  argument_list = ', '.join(p['cppName'] for p in service_helpers.expand_varargs_parameters(parameters) if not p.get("proto_only", False))
+  c_name = service_helpers.get_cname(functions, method_name, c_function_prefix)
+%>\
+% if not service_helpers.is_private_method(f) and not service_helpers.is_restricted_driver_service(config):
```
More of a curiosity but is there a reason to not include the private methods for these compilation tests?

DavidCurtiss replied:
1. This is the same filter that library.cpp was using.
2. At least some of the private methods are implemented by us, not the driver.
